job/server/src/main/java/alluxio/master/AlluxioJobMasterProcess.java

```diff
@@ -117,17 +123,38 @@ public InetSocketAddress getRpcAddress() {
     return mRpcConnectAddress;
   }
 
-  /**
-   * Starts the Alluxio job master server.
-   *
-   * @throws Exception if starting the master fails
-   */
   @Override
   public void start() throws Exception {
     mJournalSystem.start();
-    mJournalSystem.gainPrimacy();
-    startMaster(true);
-    startServing();
```

Comment:
where did these go?

Reply:
Are you relying on the fact that the state is STANDBY here and waiting to transit to PRIMARY then start the components? If so let's have some comments because the logic here is a bit confusing.